src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -122,15 +124,15 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
                         // and also, per ABNF, a single quote within a string literal is "encoded" as two consecutive single quotes in either
                         // literal or percent - encoded representation.
                         // Sample: ['a''bc','''def','xyz'''] ==> ["a'bc","'def","xyz'"], which is legitimate Json format.
-                        bracketLiteralText = NormalizeStringCollectionItems(bracketLiteralText);
+                        bracketLiteralText = NormalizeStringCollectionItems(bracketLiteralText.ToString());
                     }
                     else if (expectedTypeFullName.Equals("Edm.Guid", StringComparison.Ordinal))
                     {
                         // For collection of Guids, need to convert the Guid literals to single-quoted form, so that it is compatible
                         // with the Json reader used for deserialization.
                         // Sample: [D01663CF-EB21-4A0E-88E0-361C10ACE7FD, 492CF54A-84C9-490C-A7A4-B5010FAD8104]
                         //    ==>  ['D01663CF-EB21-4A0E-88E0-361C10ACE7FD', '492CF54A-84C9-490C-A7A4-B5010FAD8104']
-                        bracketLiteralText = NormalizeGuidCollectionItems(bracketLiteralText);
+                        bracketLiteralText = NormalizeGuidCollectionItems(bracketLiteralText.ToString());

```

Comment:
Similar to: https://github.com/OData/odata.net/pull/3122/files#r1850669734

Reply:
Used `string.Create` instead